cpu/stm32/stmclk/stmclk_l5.c

```diff
@@ -0,0 +1,359 @@
+    }
+
+    /* Enable the PLL clock only when it's used */
+    if (IS_ACTIVE(CLOCK_ENABLE_PLL)) {
+        if (IS_ACTIVE(CONFIG_CLOCK_PLL_SRC_MSI) && IS_ACTIVE(CONFIG_BOARD_HAS_LSE)) {
+            /* configure the low speed clock domain */
+            stmclk_enable_lfclk();
+            /* now we can enable the MSI PLL mode to enhance accuracy of the MSI */
+            RCC->CR |= RCC_CR_MSIPLLEN;
+            while (!(RCC->CR & RCC_CR_MSIRDY)) {}
+        }
+
+        /* configure and start the PLL */
+        RCC->PLLCFGR = (PLL_SRC | PLL_M | PLL_N | PLL_R | PLL_Q);
+
+        if (IS_ACTIVE(CONFIG_USE_CLOCK_PLL)) {
+            /* Enable PLLCLK if PLL is used as system clock */
+            RCC->PLLCFGR |= RCC_PLLCFGR_PLLREN;
+        }
+
+        if (IS_ACTIVE(CLOCK48MHZ_USE_PLLQ)) {
+            /* Enable PLLQ if PLL is used as 48MHz source clock */
+            RCC->PLLCFGR |= RCC_PLLCFGR_PLLQEN;
+        }
+
+        RCC->CR |= (RCC_CR_PLLON);
+        while (!(RCC->CR & RCC_CR_PLLRDY)) {}
+    }
+
+    /* Configure SYSCLK */
+    if (!IS_ACTIVE(CONFIG_USE_CLOCK_HSI)) {
+        RCC->CFGR &= ~RCC_CFGR_SW;
+    }
+
+    if (IS_ACTIVE(CONFIG_USE_CLOCK_HSE)) {
+        /* Select HSE as system clock */
+        RCC->CFGR |= RCC_CFGR_SW_HSE;
+        while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_HSE) {}
+    }
+    else if (IS_ACTIVE(CONFIG_USE_CLOCK_MSI)) {
+        /* Select MSI as system clock */
+        RCC->CFGR |= RCC_CFGR_SW_MSI;
+        while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_MSI) {}
+    }
+    else if (IS_ACTIVE(CONFIG_USE_CLOCK_PLL)) {
+        if (CLOCK_AHB > MHZ(80)) {
+            /* Divide HCLK by 2 before enabling the PLL */
+            RCC->CFGR |= RCC_CFGR_HPRE_3;
+        }
+
+        /* Select main PLL as system clock */
+        RCC->CFGR |= RCC_CFGR_SW_PLL;
+        while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_PLL) {}
+
+        if (CLOCK_AHB > MHZ(80)) {
+            /* Wait 1us before switching back to full speed */
+            /* Use volatile to prevent the compiler from optimizing the loop */
+            volatile uint8_t count = CLOCK_CORECLOCK / MHZ(1);
+            while (count--) {}
+            RCC->CFGR &= ~RCC_CFGR_HPRE_3;
+        }
+    }
+
+    if (!IS_ACTIVE(CLOCK_ENABLE_HSI)) {
+        /* Disable HSI only if not used */
+        stmclk_disable_hsi();
+    }
+
+    if (IS_ACTIVE(CLOCK_ENABLE_48MHZ)) {
+        /* configure the clock used for the 48MHz clock tree (USB, RNG) */
+        RCC->CCIPR1 = CLOCK48MHZ_SELECT;
+    }
+
+    if (IS_USED(MODULE_PERIPH_RTT)) {
+        /* Ensure LPTIM1 clock source (LSI or LSE) is correctly reset when initializing
+           the clock, this is particularly useful after waking up from deep sleep */
+        if (IS_ACTIVE(CONFIG_BOARD_HAS_LSE)) {
+            RCC->CCIPR1 |= RCC_CCIPR1_LPTIM1SEL_0 | RCC_CCIPR1_LPTIM1SEL_1;
+        }
+        else {
+            RCC->CCIPR1 |= RCC_CCIPR1_LPTIM1SEL_0;
+        }
+    }
```

Comment:
same as on l4?

Reply:
RTT is not implemented for L5 by this PR so this whole block could be simply removed. Otherwise, it's similar to L4 I guess except that the register is `RCC->CCIPR1` and on L4 it's `RCC->CCIPR` (and all related defines must be changed as well...)